### DISCUSSION TITLE
What does :invocation[0] mean?

### DISCUSSION THREAD

**Discussion by nsikandar16** [statelyai/xstate#3499] [Q&A]
Hi,

Can somebody here please help in understanding `:invocation[0]`?
I'm getting it in an action's invoke done type like `done.invoke.sendEmail:invocation[0]`?

**Comment by Andarist** [comment]
It's a pseudo-random ID that has been assigned to your invocation. The reason why we have to assign something unique~ there is for events coming from different invocations to be isolated from events from similar (with the same `src`) invocations. Consider this example:
```ts
{
  type: 'parallel',
  states: {
    a: {
      invoke: {
        src: 'sendEmail',
        onDone: '#foo'
      }
    },
    b: {
      invoke: {
        src: 'sendEmail',
        onDone: '#bar'
      }
    }
  }
}
```

In this example you wouldn't like to take `a`'s transition when `b`'s service completes.
